OpenTrons/opentrons-api
# feat(step-generation): error if gripper moves when pipette has tip on it

closes RAUT-907, addresses RQA-2098

# Overview

This was an error that was reported awhile back (RQA-2098) and was since addressed in PE but we needed to address it in PD as well. So if you move the gripper when any pipette has a tip on it still, it will have a timeline error to avoid collisions

<img width="869" alt="Screen Shot 2024-02-21 at 1 34 05 PM" src="https://github.com/Opentrons/opentrons/assets/66035149/678254e0-f441-41c5-b916-5c659e23330f">

# Test Plan

1. Create a flex protocol and add the gripper
2. Add a transfer step from 1 well to another well. 
3. Add a move labware step with the gripper
4. Add another transfer step from 1 well to another well but this time, modify the the `Change Tip` option to `never`. This means that step-generation will not drop tip after the 1st transfer, therefore leaving the tip on the pipette when the move labware step occurs. 
5. observe the timeline error
6. modify the move labware step to not use the gripper and the timeline error should go away

# Changelog

- check if there is a tip on any pipette, if so and the gripper is in use, emit error.
- add new error creator for tip on pipette
- add test case

# Review requests

see test plan

# Risk assessment

low
